GameEngine/Agents/SearchNode.cs

```diff
@@ -44,7 +43,7 @@ private SearchNode ChildNode(Play play)
                 State = State.Successor(play),
                 Parent = this,
                 Action = play,
-                PathCost = PathCost + StepCost
+                PathCost = PathCost + Action?.StepCost ?? 1
```

Comment:
When would `Action` be null?

Reply:
The `RootNode` doesn't have an `Action`. Which speaks to a bug. The `PathCost` of the new node should use the `StepCost`  of `play`, not of `Action`. Fixed.